providers/common/ciphers/cipher_common.c

```diff
@@ -67,9 +69,9 @@ CIPHER_DEFAULT_GETTABLE_CTX_PARAMS_START(cipher_generic)
 CIPHER_DEFAULT_GETTABLE_CTX_PARAMS_END(cipher_generic)
 
 static const OSSL_PARAM cipher_known_settable_ctx_params[] = {
-    OSSL_PARAM_int(OSSL_CIPHER_PARAM_KEYLEN, NULL),
-    OSSL_PARAM_int(OSSL_CIPHER_PARAM_PADDING, NULL),
-    OSSL_PARAM_int(OSSL_CIPHER_PARAM_NUM, NULL),
+    OSSL_PARAM_size_t(OSSL_CIPHER_PARAM_KEYLEN, NULL),
+    OSSL_PARAM_size_t(OSSL_CIPHER_PARAM_PADDING, NULL),
```

Comment:
Again, really?  I don't think it is a length.

Reply:
What would it OSSL_PARAM_construct_???

Reply:
uint maybe?

Reply:
Ok will change blocksize, padding & mode to be uint

Reply:
Updated

Reply:
????????? `blocksize` is a size, no?  Why should you change that one?

Reply:
how big is the blocksize ever going to be?

Reply:
Isnt is normally 16?

Reply:
Is that relevant?  How bit is an IV length ever going to be?  A key length?, the MAC output size? ...

What's going on here is that we're moving along with the rest of the world and are making sizes, lengths and indexes typed `size_t` *where possible* (i.e. where patterns in public functions allow us...)

Reply:
My comments were about the padding and mode, not the block size....

I agree that blocksize should be `size_t`.

Reply:
Ok updated...